integration/e2e_test.go

```diff
@@ -0,0 +1,289 @@
+			} else {
+				attempt++
+				t.Log("Retrying")
+				time.Sleep(10 * time.Second)
+			}
+		} else {
+			machines = strings.Count(output_str, "\n")
+			running := strings.Count(output_str, "Running")
+			t.Logf("Machines %d out of which %d are Running", machines, running)
+			if machines == running {
+				break
+			} else {
+				attempt++
+				time.Sleep(10 * time.Second)
+				t.Log("Waiting for machines to start running")
+			}
+		}
+	}
+
+	// Make sure we have as many nodes as machines
+	attempt = 0
+	max_attempts = 60
+	command_str = fmt.Sprintf("kubectl --kubeconfig=%s get no --no-headers", KUBECONFIG)
+	command = strings.Fields(command_str)
+	for {
+		cmd = exec.Command(command[0], command[1:]...)
+		output, err = cmd.CombinedOutput()
+		output_str := string(output)
+		if err != nil {
+			t.Log(output_str)
+			if attempt >= max_attempts {
+				t.Fatal(err)
+			} else {
+				attempt++
+				time.Sleep(10 * time.Second)
+				t.Log("Retrying")
+			}
+		} else {
+			nodes := strings.Count(output_str, "\n")
+			ready := strings.Count(output_str, " Ready")
+			t.Logf("Machines are %d, Nodes are %d out of which %d are Ready", machines, nodes, ready)
+			if machines == nodes && ready == nodes {
+				break
+			} else {
+				attempt++
+				time.Sleep(20 * time.Second)
+				t.Log("Waiting for nodes to become ready")
+			}
+		}
+	}
+}
+
+func deployMicrobot(t testing.TB) {
+	t.Log("Deploying microbot")
+	command_str := fmt.Sprintf("kubectl --kubeconfig=%s delete deployment bot", KUBECONFIG)
+	command := strings.Fields(command_str)
+	cmd := exec.Command(command[0], command[1:]...)
+	cmd.Run()
+
+	command_str = fmt.Sprintf("kubectl --kubeconfig=%s create deploy --image=cdkbot/microbot:1 --replicas=30 bot", KUBECONFIG)
+	command = strings.Fields(command_str)
+	cmd = exec.Command(command[0], command[1:]...)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Error(string(output))
+		t.Fatalf("Failed to create the requested microbot deployment. %s", err)
+	}
+
+	// Make sure we have as many nodes as machines
+	attempt := 0
+	max_attempts := 60
+	t.Log("Waiting for the deployment to complete")
+	command_str = fmt.Sprintf("kubectl --kubeconfig=%s wait deploy/bot --for=jsonpath={.status.readyReplicas}=30", KUBECONFIG)
+	command = strings.Fields(command_str)
+	for {
+		cmd = exec.Command(command[0], command[1:]...)
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			t.Log(string(output))
+			if attempt >= max_attempts {
+				t.Fatal(err)
+			} else {
+				attempt++
+				t.Log("Retrying")
+				time.Sleep(10 * time.Second)
+			}
+		} else {
+			break
+		}
+	}
+}
+
+// TestBasic waits for the target cluster to deploy and start a 30 pod deployment.
+// The CLUSTER_MANIFEST_FILE environment variable should point to a manifest with the target cluster
+// kubectl and clusterctl have to be avaibale in the caller's path.
+// kubectl should be setup so it uses the kubeconfig of the management cluster by default.
+func TestBasic(t *testing.T) {
```

Comment:
nit: Shall we put the public function at the top of all the private functions?
LGTM otherwise.